File: CRM-FRONT/crm/src/services/dailySalesService.js
import axiosInstance from "../utils/axiosInstance";

const dailySalesService = {
  create: (data) => axiosInstance.post("/daily-sales", data),
  getById: (id) => axiosInstance.get(`/daily-sales/${id}`),
  update: (id, data) => axiosInstance.put(`/daily-sales/${id}`, data),
  delete: (id) => axiosInstance.delete(`/daily-sales/${id}`),
  search: ({ page = 0, size = 50, ...criteria }) =>
    axiosInstance.get("/daily-sales", {
      params: { page, size, ...criteria },
    }),
};

export default dailySalesService;

File: product/src/main/java/com/crm/product/service/DailySalesService.java
package com.crm.product.service;

import com.crm.product.entities.dto.SearchDailySalesCriteria;
import com.crm.product.entities.dto.request.DailySalesRequestDTO;
import com.crm.product.entities.dto.request.DailySalesUpdateRequestDTO;
import com.crm.product.entities.dto.response.DailySalesResponseDTO;
import com.crm.product.entities.dto.response.DailySalesUpdateResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface DailySalesService {
    DailySalesResponseDTO create(DailySalesRequestDTO dto);
    DailySalesResponseDTO getById(String id);
    Page<DailySalesResponseDTO> search(SearchDailySalesCriteria criteria, Pageable pageable);
    void delete(String id);
    DailySalesUpdateResponseDTO update(String id, DailySalesUpdateRequestDTO dto);
}
